src/UnityExtension/Assets/Editor/GitHub.Unity/UpdateCheck.cs

```diff
@@ -0,0 +1,230 @@
+﻿using System;
+using UnityEngine;
+using UnityEditor;
+using System.Linq;
+using GitHub.Logging;
+
+namespace GitHub.Unity
+{
+    [Serializable]
+    class GUIPackage
+    {
+        [SerializeField] private string version;
+        [SerializeField] private string url;
+        [SerializeField] private string releaseNotes;
+        [SerializeField] private string releaseNotesUrl;
+        [SerializeField] private string message;
+
+        [NonSerialized] private Package package;
+        public Package Package
+        {
+            get
+            {
+                if (package == null)
+                {
+                    package = new Package
+                    {
+                        Version = TheVersion.Parse(version),
+                        Url = url,
+                        ReleaseNotes = releaseNotes,
+                        ReleaseNotesUrl = releaseNotesUrl,
+                        Message = message
+                    };
+                }
+                return package;
+            }
+        }
+
+        public GUIPackage()
+        {}
+
+        public GUIPackage(Package package)
+        {
+            version = package.Version.ToString();
+            url = package.Url;
+            releaseNotes = package.ReleaseNotes;
+            releaseNotesUrl = package.ReleaseNotesUrl;
+            message = package.Message;
+        }
+    }
+
+    public class UpdateCheckWindow :  EditorWindow
+    {
+        public const string UpdateFeedUrl =
+#if DEBUG
+        "http://localhost:50000/unity/latest.json"
+#else
+        "https://ghfvs-installer.github.com/unity/latest.json"
+#endif
+        ;
+
+        public static void CheckForUpdates()
+        {
+            var download = new DownloadTask(TaskManager.Instance.Token, EntryPoint.Environment.FileSystem, UpdateFeedUrl, EntryPoint.Environment.UserCachePath);
+            download.OnEnd += (thisTask, result, success, exception) =>
+            {
+                if (success)
+                {
+                    try
+                    {
+                        Package package = result.ReadAllText().FromJson<Package>(lowerCase: true, onlyPublic: false);
+                        TheVersion current = TheVersion.Parse(ApplicationInfo.Version);
+                        TheVersion newVersion = package.Version;
+
+                        TheVersion[] versionsToSkip = EntryPoint.ApplicationManager.UserSettings.Get<TheVersion[]>(Constants.SkipVersionKey);
```

Comment:
Why did you choose to store an array and not the last skipped version?

Reply:
Good point, the latest skipped version is enough